src/_C014.ino

```diff
@@ -0,0 +1,686 @@
+          CPlugin_014_sendMQTTdevice(pubname, "$nodes", nodesList.c_str(), errorCounter);
+        }
+
+        log = F("C014 : autodiscover information of ");
+        log += deviceCount;
+        log += F(" Devices and ");
+        log += nodeCount;
+        log += F(" Nodes sent with ");
+        if (errorCounter>0)
+        {
+          // alert: this is the state the device is when connected to the MQTT broker, but something wrong is happening. E.g. a sensor is not providing data and needs human intervention. You have to send this message when something is wrong.
+          CPlugin_014_sendMQTTdevice(pubname,"$state","alert",errorCounter);
+          log+=errorCounter;
+        } else {
+          // ready: this is the state the device is in when it is connected to the MQTT broker, has sent all Homie messages and is ready to operate. You have to send this message after all other announcements message have been sent.
+          CPlugin_014_sendMQTTdevice(pubname,"$state","ready",errorCounter);
+          log+=F("no");
+        }
+        log+=F(" errors! (");
+        log+=msgCounter;
+        log+=F(" messages)");
+        msgCounter = 0;
+        errorCounter = 0;
+        addLog(LOG_LEVEL_DEBUG, log);
+
+        break;
+      }
+
+    case CPLUGIN_PROTOCOL_TEMPLATE:
+      {
+        event->String1 = F(CPLUGIN_014_SUBSCRIBE);
+        event->String2 = F(CPLUGIN_014_PUBLISH);
+        break;
+      }
+
+    case CPLUGIN_DICONNECT:
+      {
+        pubname = CPLUGIN_014_BASE_TOPIC; // Scheme to form device messages
+        pubname.replace(F("%sysname%"), Settings.Name);
+        // disconnected: this is the state the device is in when it is cleanly disconnected from the MQTT broker. You must send this message before cleanly disconnecting
+        CPlugin_014_sendMQTTdevice(pubname,"$state","disconnected",errorCounter);
+        break;
+      }
+
+    case CPLUGIN_FALLING_ASLEEP:
+      {
+        pubname = CPLUGIN_014_BASE_TOPIC; // Scheme to form device messages
+        pubname.replace(F("%sysname%"), Settings.Name);
+        // sleeping: this is the state the device is in when the device is sleeping. You have to send this message before sleeping.
+        CPlugin_014_sendMQTTdevice(pubname,"$state","sleeping",errorCounter);
+        break;
+      }
+
+    case CPLUGIN_PROTOCOL_RECV:
+      {
+        byte ControllerID = findFirstEnabledControllerWithId(CPLUGIN_ID_014);
+        if (ControllerID == CONTROLLER_MAX) {
+          // Controller is not enabled.
+          break;
+        } else {
+          String cmd;
+          struct EventStruct TempEvent;
+          TempEvent.TaskIndex = event->TaskIndex;
+          TempEvent.Source = VALUE_SOURCE_MQTT; // to trigger the correct feedback
+          bool validTopic = false;
+          int lastindex = event->String1.lastIndexOf('/');
+          const String lastPartTopic = event->String1.substring(lastindex + 1);
```

Comment:
This `lastPartTopic` is created, but does not seem to be used?

Reply:
same "legacy" as before.